sample-app/src/components/DropdownSection.tsx

```diff
@@ -59,30 +60,31 @@ export default function DropdownSection({
   }
 
   function handleKeyDown(evt: globalThis.KeyboardEvent) {
-    if (focusStatus === 'active') {
-      if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(evt.key)) {
-        evt.preventDefault();
-      }
-
-      if (evt.key === 'ArrowDown' || evt.key === 'ArrowRight') {
-        incrementOptionFocus();
-      } else if (evt.key === 'ArrowUp' || evt.key === 'ArrowLeft') {
-        decrementOptionFocus();
-      }
+    if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(evt.key)) {
+      evt.preventDefault();
     }
-  }
 
-  useEffect(() => {
-    if (focusStatus === 'active') {
-      onFocusChange(options[focusedOptionIndex].value, `${optionIdPrefix}-${focusedOptionIndex}`);
+    if (evt.key === 'ArrowDown' || evt.key === 'ArrowRight') {
+      incrementOptionFocus();
+    } else if (evt.key === 'ArrowUp' || evt.key === 'ArrowLeft') {
+      decrementOptionFocus();
+    } else if (evt.key === 'Enter') {
+      onSelect(options[focusedOptionIndex].value, focusedOptionIndex);
     }
-  }, [focusStatus, focusedOptionIndex, onFocusChange, optionIdPrefix, options]);
+  }
 
   useEffect(() => {
     if (focusStatus === 'active') {
+      if (!wasActive.current) {
```

Comment:
why did we need to combine these two effects? I also feel like we could have pass dependencies to the addEventListener effect so we're not adding and removing event listeners unnecessarily

Reply:
By adding the useRef, there wasn't anything differentiating the two useEffects because we're relying on the ref rather than a dependency to determine when the focus has changed. So it was simpler to combine them.

For the event listeners, if we pass dependencies, the other 3 functions (`handleKeyDown`, `increment/decrementOptionFocus`) would need to be wrapped in useCallbacks. I can do that if we prefer to be on the safe side, but right now, this useEffect isn't causing additional adding/removing of the listeners because any time this component would have to re-render, InputDropdown also needs to re-render. And when InputDropdown re-renders, these sections are unmounted and all the children sections are re-mounted anyways.